src/backend/distributed/operations/stage_protocol.c

```diff
@@ -380,7 +380,13 @@ citus_update_table_statistics(PG_FUNCTION_ARGS)
 
 	Oid distributedTableId = PG_GETARG_OID(0);
 
-	UpdateTableStatistics(distributedTableId);
+	Relation relation = try_relation_open(distributedTableId, AccessShareLock);
```

Comment:
Why do we use `try_relation_open`? What happens if we cannot acquire the lock?

Also, I think this needs an isolation test given the issues we had

Reply:
With this command `select citus_update_table_statistics(logicalrelid) from pg_dist_partition;` we will call the function for each table one by one and if the table has already been dropped, using `relation_open` will give an error saying the table doesn't exist.

Reply:
As far as I know `select citus_update_table_statistics(logicalrelid) from pg_dist_partition;` is called with the rebalancer page load on cloud, do we know what happens if the call cannot finish because we can't take the lock? I guess they use a timeout?